Make every configuration optional (fixes #42)

leplatrem opened the pull request:
fixes #42 

While wrinting the documentation, I realize that configuration could have default everywhere

Natim commented on cliquet/__init__.py:
```diff
@@ -154,14 +156,16 @@ def includeme(config):
 
-    config.registry.project_name = settings['cliquet.project_name']
-    config.registry.project_docs = settings['cliquet.project_docs']
+    config.registry.project_name = settings.get('cliquet.project_name', '')
```
Not sure for this one.

Natim replied:
I mean the project_name configuration should be mandatory and not default to empty string.

leplatrem replied:
Why ? it is used in hello view only !

Default to cliquet ?

ametaireau replied:
Aim is to fail if it's not defined so that cliquet users define it. We can also fallback to a default value but in this case issuing a warning would help.

Natim replied:
Maybe we should do the same thing as we do with version for the default value, use the default package name.

ametaireau replied:
We don't do that anymore in cliquet: cliquet doesn't make any assumptions about the version, and the client must provide it explicitely.

Natim replied:
I mean passing it as we do for version rather than reading it from the conf

ametaireau replied:
Ah okay, I understand your proposal I think. Let me rephrase so I'm sure we're on the same page here.

Since we're already passing the version information to cliquet in `initialize_cliquet`, we should reuse this version to display it on the home page.

If that's it, I concur :)

Natim replied:
> we should reuse this version to display it on the home page.
Isn't it already the case?

ametaireau replied:
Hmm, I'm almost all wrong. We're already doing that (I wrote the patch for it, meh).

Your proposal is to do the same thing for the name. If yes, then I believe we should use that for default, but we should let the possibility to the user to configure another name (for instance,  "cloud storage" is a project name and "kinto" is the codename. They aren't the same).

Natim replied:
> Your proposal is to do the same thing for the name. If yes, then I believe we should use that for default

Totally agree :+1:

Natim commented on cliquet/__init__.py:
```diff
@@ -19,6 +19,8 @@
 from cliquet.utils import msec_time
 
 DEFAULT_OAUTH_CACHE_SECONDS = 5 * 60
+DEFAULT_STORAGE_BACKEND = 'cliquet.storage.postgresql'
```
Not sure if the default should be postgresql

leplatrem replied:
ah yes, and that would mean it should  not be in extra_require then !

Natim commented:
Refs https://github.com/mozilla-services/readinglist/issues/81

leplatrem commented:
r?

leplatrem commented:
If you agree, I update this PR to include #42 and implement what you suggest about project name

Natim commented:
Yes ok.

Natim commented:
A warning is not exactly what I suggested about project name :)

leplatrem commented:
Yes I know

leplatrem commented:
@ametaireau @Natim r?

Here is what I did: 
* Read settings from environment
* Centralize defaults
* Cleaned tests settings to use defaults
* Got rid of ``settings.get`` since I assume we have defaults for everything
* Got rid of registry usage for hello view value (*thought it has better consistency*)
* Added project_name to initialize_cliquet (*not a fan of this btw, I would have liked to keep this method very very stupid*)
* Warn if no project name
* Raise error is no project version (can also be specified in settings)
